### ISSUE TITLE
boost::uuids::entropy_error in 11_1_X after  #5573 (update to boost  v1.72.0)

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmsdist#5657]
I see problems after  #5573 was merged in workflows running PF reco 

I incidentally found the same problem as was already seen for a while on PPC.
My setup is on a native SL6 node running a cc7 container and see the same problem

```
 An exception of category 'StdException' occurred while
    [0] Processing  stream begin Run run: 152698 stream: 0
    [1] Calling method for module PFProducer/'particleFlowTmp'
    [2] Using EventSetup component PoolDBESSource/'GlobalTag' to make data PerformancePayload/'' in record PFCalibrationRcd
 Exception Message:
 A std::exception was thrown.
 getrandom
```

gdb stack trace shows 
```
#1  0x00007fc6c97f49d8 in void boost::throw_exception<boost::exception_detail::error_info_injector<boost::uuids::entropy_error> >(boost::exception_detail::error_info_injector<boost::uuids::entropy_error> const&) () from /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_1_X_2020-03-09-1100/lib/slc7_amd64_gcc820/libCondFormatsPhysicsToolsObjects.so
#2  0x00007fc6c97f1859 in PerformancePayloadFromTFormula::initialize() ()
```
This is apparently more specifically coming from 
https://github.com/cms-sw/cmssw/blob/ecf11e6c9bc34871e16a53e34ed4b8e750e2b3f5/CondFormats/PhysicsToolsObjects/src/PerformancePayloadFromTFormula.cc#L14-L19
```c++
    boost::uuids::random_generator gen;
    boost::uuids::uuid uniqueFormulaId = gen();
```

**Comment by slava77** [comment]
perhaps this is relevant https://github.com/boostorg/uuid/issues/79

```BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX```

**Comment by smuzaffar** [comment]
@slava77 , please trying setting `BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX` in your config/toolbox/slc7_amd64_gcc820/tools/selected/boost.xml and run
```
scram build checkdeps
scram build -j 8
```
and retry the workflow (or let me know how to re-produce the error).

**Comment by smuzaffar** [comment]
ok I see that workflow 140.51 for PPC is failing with such an error. I will it to see if the above suggestion works.

**Comment by slava77** [comment]
> and retry the workflow (or let me know how to re-produce the error).

from problems in the jenkins tests I know that you should get the same error on the PPC by rerunning wf 140.52.

on x86, do we still have commonly accessible SL6 machines? I can check if I can reproduce the same issue there.

**Comment by smuzaffar** [comment]
we still have lxplus6 access but I am not able to run singularity there. Anyway, I will try it for PPC

**Comment by slava77** [comment]
> scram build checkdeps
> scram build -j 8

with boost it will take a while

**Comment by smuzaffar** [comment]
yes it will nearly re-build full cmssw

**Comment by slava77** [comment]
adding ```BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX``` helped (unless the reason was that I built this from inside the container where I do the test). After a rebuild I do not get a crash anymore.

**Comment by smuzaffar** [comment]
@slava77 , this wf is failing on our ppc64 machine too. We build and run on the same machine. so I do not think that system dynamically picks different getrandom implementation. Anyway, adding `BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX` and re-building just `CondFormats/PhysicsToolsObjects` fixed the issue for ppc64. 

Question is should we add this flag or should drop the usage of `boost::uuids::random_generator` ? It is only used in 3 places https://github.com/cms-sw/cmssw/search?q=random_generator&type=

**Comment by slava77** [comment]
I was perhaps too fast to make #5658
I agree that an alternative solution is to get rid of ```random_generator``` use.
A replacement would be something else that generates a random string

**Comment by slava77** [comment]
well, what would be the downside of doing ```BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX``` ?

I'm thinking of why don't we do both this and (eventually) get rid of ```boost::uuids::random_generator```

**Comment by slava77** [comment]
grepping some more, it looks like we have an in-house alternative ```edm::createGlobalIdentifier()```

**Comment by slava77** [comment]
@smuzaffar 
I guess this can be closed now after #5658 was merged
